ts/project-details.ts

```diff
@@ -74,8 +192,54 @@ function reload(definition: TableDefinition<any, any>) {
     handleFilters
   ).build();
 }
-
 function handleFilters(key: string, value: string, add?: boolean) {
   projectSummaryTable.applyFilters(key, value, add);
   updateUrlParams(key, value, add);
 }
+
+// reloadSummary: destroy current table and build new project summary table
+function reloadSummary() {
+  projectSummaryContainer?.firstChild?.remove();
+  if (tableContainer?.dataset.detailValue) {
+    projectSummaryTable = new TableBuilder(
+      getProjectSummaryRowDefinition(
+        urls.rest.projects.summary(tableContainer.dataset.detailValue)
+      ),
+      "projectSummaryContainer",
+      getSearchParams()
+    ).build();
+  }
+}
+
+function handleDateFilter(filterKey: string, filterValue: string) {
+  // update URL to include query parameters
+  replaceUrlParams(filterKey, filterValue);
+  reloadSummary();
+}
+
+function handleDateDropdownFilter(filterKey: string, filterValue: string) {
+  // update URL to include query parameters
+  replaceUrlParams(filterKey, filterValue);
+  if (filterKey == "AFTER_DATE") {
+    const beforeDateCOntainer = document.getElementById(
+      "afterDate"
+    ) as HTMLInputElement;
+    beforeDateCOntainer.value = filterValue;
+  } else {
+    const afterDateContainer = document.getElementById(
+      "beforeDate"
+    ) as HTMLInputElement;
+    afterDateContainer.value = filterValue;
+  }
+  reloadSummary();
```

Comment:
Can the table builder's `applyFilters` method be used instead of rebuilding the table entirely?

The pattern of calling `handleDateDropdownFilter` twice when choosing a range from the dropdown is causing two requests to happen, which isn't ideal. Maybe table builder's `applyFilters` should be renamed to `applyFilter`, and there should be another `applyFilters` method that accepts multiple filters to apply at once?